src/ui/setup/dynamic/devtools.ts

```diff
@@ -152,6 +153,15 @@ export default async function DevTools(store: AppStore) {
   window.app.debugger = setupDebuggerHelper();
   window.app.prefs = window.app.prefs ?? {};
 
+  if (!replayClient) {
+    // This _should_ have been passed in from `RecordingPage`.
+    // But, some test code also calls `setupDevtools()`, and those
+    // test files don't call `useReplayClient` anywhere.
+    // Grab the usual instance manually so we can pass it to thunks.
+    const replayClientModule = await import("shared/client/ReplayClientContext");
+    replayClient = replayClientModule.replayClient;
+  }
```

Comment:
`setupDevtools()` is called in some of the unit-ish tests, and I was trying to make sure that at least _some_ instance of `replayClient` was available in case that logic ended up trying to access it.  It may not be 100% necessary - I can try running the tests without it and see what happens.

Reply:
I'd be curious to know what you found here.

Reply:
Just tried removing that, and sure enough, several tests fail:

```
 FAIL  src/devtools/client/webconsole/components/__tests__/WebConsoleApp.test.tsx (12.992 s)
  ● Web Console UI › Can render a list of messages and filter it

    TypeError: Cannot read properties of undefined (reading 'findKeyboardEvents')

      73 |
      74 |     replayClient
    > 75 |       .findKeyboardEvents(({ events }) => onKeyboardEvents(events, store))
         |        ^
      76 |       .then(() => {
      77 |         store.dispatch(loadReceivedKeyboardEvents());
      78 |       });

      at findKeyboardEvents (src/ui/actions/app.ts:75:8)
```